Sources/FoundationEssentials/JSON/JSONEncoder.swift

```diff
@@ -1214,9 +1214,9 @@ private extension __JSONEncoder {
             return self.wrap(url.absoluteString)
         } else if let decimal = value as? Decimal {
             return .number(decimal.description)
-        } else if let encodable = value as? _JSONStringDictionaryEncodableMarker {
+        } else if !options.keyEncodingStrategy.isDefault, let encodable = value as? _JSONStringDictionaryEncodableMarker {
             return try self.wrap(encodable as! [String:Encodable], for: additionalKey)
```

Comment:
Are we able to use `_specializingCast` here as well?

Reply:
Unfortunately, no, we can't use it here. `_specializingCast` internally performs type equality check, and here we use `as?` to check protocol conformance. so `_specializingCast` is inapplicable here